extensions/amp-video/0.1/amp-video.js

```diff
@@ -83,18 +90,47 @@ class AmpVideo extends AMP.BaseElement {
    * @override
    */
   preconnectCallback(opt_onLayout) {
-    const videoSrc = this.getVideoSource_();
+    const videoSrc = this.getVideoSourceForPreconnect_();
     if (videoSrc) {
       assertHttpsUrl(videoSrc, this.element);
       this.preconnect.url(videoSrc, opt_onLayout);
     }
   }
 
+  /**
+   * @override
+   *
+   * @overview
+   * AMP Cache may selectively cache certain video sources (based on size, type
+   * etc..). When AMP Cache does so, it rewrites the `src` for `amp-video` and
+   * `source` children that are cached and adds a `amp-orig-src` attribute
+   * pointing to the original source.
+   *
+   * There are two separate runtime concerns that we handle here:
+   *
+   * 1) Handling 404s
+   * Eventhough AMP Cache rewrites the `src` to point to the CDN, the actual
+   * video may not be ready in the cache yet, in those cases the CDN will
+   * return a 404. Runtime handles this situation by appending an additional
```

Comment:
Other cases for 404 are if the video type is not supported but the URL was rewritten anyway or if it's too large.